File: benchmarks/python/string_processing.py
#!/usr/bin/env python3
"""BENCH_STRING_ITERS=3000 BENCH_STRING_CHUNK=benchmark-chunk-0123456789"""


def build_string(iters: int, chunk: str) -> int:
    out = ""
    for _ in range(iters):
        out += chunk
    return len(out)


def main() -> None:
    print(build_string(3000, "benchmark-chunk-0123456789"))


if __name__ == "__main__":
    main()

File: benchmarks/rust/src/bin/string_processing.rs
fn build_string(iters: i32, chunk: &str) -> usize {
    let mut out = String::new();
    for _ in 0..iters {
        out.push_str(chunk);
    }
    out.len()
}

fn main() {
    println!("{}", build_string(3000, "benchmark-chunk-0123456789"));
}
